geth/node/manager.go

```diff
@@ -563,3 +565,16 @@ func (m *NodeManager) RPCServer() (*rpc.Server, error) {
 
 	return m.rpcServer, nil
 }
+
+// isNodeAvailable check if we have a node running and make sure is fully started
+func (m *NodeManager) isNodeAvailable() error {
+	if m.nodeStarted == nil {
+		return ErrNoRunningNode
+	}
+
+	if m.node == nil {
+		return ErrNodeOffline
```

Comment:
Hm. I may be wrong but I believe both these cases mean `ErrNoRunningNode`.

Reply:
Yeah, I was wondering about this one too. I am not sure we can tell if a node is offline from here. Even if there is no Internet access `m.node` probably is not `nil`. I will check that as I guess it was not verified.

Reply:
Follow up: https://github.com/status-im/status-go/issues/242#issuecomment-326265578

Reply:
After testing against the current develop using iOS simulator: https://github.com/status-im/status-react/issues/1295#issuecomment-327140478